src/lib/peer.js

```diff
@@ -59,12 +61,22 @@ function Peer(config) {
             self.fingerTable = new FingerTable(self.peerId, 
                                                self.events,
                                                self.channelManager);
+            var readyEventSent = false;
+            self.fingerTable.events.on('fingerUpdate', function(data){
+                if(readyEventSent) return;
+                //Send the ready event for the first 'fingerUpdate' event
+                readyEventSent = true;
+                self.events.emit('ready', {});
+            });
             self.events.emit('registered', {peerId: data.peerId});
         }
 
-        ioc.emit('s-register', {});
+        ioc.emit('s-register', {'id' : peerId});
```

Comment:
I have some modified requirements on the framework, one among is providing a way to give the signalling server a hint which id the client wants to have.
I think that's just one of the changes you should ignore ;-)
Maybe I should have made a seperate repository with these changes, sorry for that